src/heliaServer.ts

```diff
@@ -219,13 +220,11 @@ export class HeliaServer {
    * Convert HeliaFetchOptions to a URL string compatible with `@helia/verified-fetch`
    */
   #getUrlFromArgs ({ address, namespace, relativePath, peerId, cid }: HeliaFetchOptions): string {
-    // if (peerId != null) {
-    //   // we have a peerId, so we need to pass `ipns://<peerId>/relativePath` to @helia/verified-fetch.
-    //   return `${namespace}://${peerId.toString()}/${relativePath}`
-    // }
-    if (relativePath == null || relativePath === '' || relativePath === '/') {
-      // we have a CID, so we need to pass `ipfs://<cid>` to @helia/verified-fetch.
-      return `${namespace}://${address}`
+    // The if a DNSLink returned here needs to be encoded per https://specs.ipfs.tech/http-gateways/subdomain-gateway/#host-request-header
+    if (namespace === 'ipns') {
+      if (isDnsLabel(address)) {
+        address = dnsLinkLabelDecoder(address)
+      }
```

Comment:
@SgtPooki  small nit: We should check if `address` is a valid CID (with [ipns name](https://specs.ipfs.tech/ipns/ipns-record/#ipns-name)) first, and apply this normalization only if it is _not_ a valid CID.

(better safe than sorry, there could be a multibase encoding which uses `.` and `-`)